libcrux-ml-dsa/src/simd/portable/encoding/t0.rs

```diff
@@ -81,50 +78,52 @@ pub fn deserialize(serialized: &[u8]) -> PortableSIMDUnit {
     let byte11 = serialized[11] as i32;
     let byte12 = serialized[12] as i32;
 
-    simd_unit.coefficients[0] = byte0;
-    simd_unit.coefficients[0] |= byte1 << 8;
-    simd_unit.coefficients[0] &= BITS_IN_LOWER_PART_OF_T_MASK;
-
-    simd_unit.coefficients[1] = byte1 >> 5;
-    simd_unit.coefficients[1] |= byte2 << 3;
-    simd_unit.coefficients[1] |= byte3 << 11;
-    simd_unit.coefficients[1] &= BITS_IN_LOWER_PART_OF_T_MASK;
-
-    simd_unit.coefficients[2] = byte3 >> 2;
-    simd_unit.coefficients[2] |= byte4 << 6;
-    simd_unit.coefficients[2] &= BITS_IN_LOWER_PART_OF_T_MASK;
-
-    simd_unit.coefficients[3] = byte4 >> 7;
-    simd_unit.coefficients[3] |= byte5 << 1;
-    simd_unit.coefficients[3] |= byte6 << 9;
-    simd_unit.coefficients[3] &= BITS_IN_LOWER_PART_OF_T_MASK;
-
-    simd_unit.coefficients[4] = byte6 >> 4;
-    simd_unit.coefficients[4] |= byte7 << 4;
-    simd_unit.coefficients[4] |= byte8 << 12;
-    simd_unit.coefficients[4] &= BITS_IN_LOWER_PART_OF_T_MASK;
-
-    simd_unit.coefficients[5] = byte8 >> 1;
-    simd_unit.coefficients[5] |= byte9 << 7;
-    simd_unit.coefficients[5] &= BITS_IN_LOWER_PART_OF_T_MASK;
-
-    simd_unit.coefficients[6] = byte9 >> 6;
-    simd_unit.coefficients[6] |= byte10 << 2;
-    simd_unit.coefficients[6] |= byte11 << 10;
-    simd_unit.coefficients[6] &= BITS_IN_LOWER_PART_OF_T_MASK;
-
-    simd_unit.coefficients[7] = byte11 >> 3;
-    simd_unit.coefficients[7] |= byte12 << 5;
-    simd_unit.coefficients[7] &= BITS_IN_LOWER_PART_OF_T_MASK;
-
-    simd_unit.coefficients[0] = change_t0_interval(simd_unit.coefficients[0]);
-    simd_unit.coefficients[1] = change_t0_interval(simd_unit.coefficients[1]);
-    simd_unit.coefficients[2] = change_t0_interval(simd_unit.coefficients[2]);
-    simd_unit.coefficients[3] = change_t0_interval(simd_unit.coefficients[3]);
-    simd_unit.coefficients[4] = change_t0_interval(simd_unit.coefficients[4]);
-    simd_unit.coefficients[5] = change_t0_interval(simd_unit.coefficients[5]);
-    simd_unit.coefficients[6] = change_t0_interval(simd_unit.coefficients[6]);
-    simd_unit.coefficients[7] = change_t0_interval(simd_unit.coefficients[7]);
+    let mut coefficient0 = byte0;
```

Comment:
Can't you put the `mut` to the `byte0`? Then we don't need this line every time.

Reply:
It is tempting and I tried it, but the `byte` values are distributed across different coefficients, so there isn't a 1-1 mapping between `coefficienti` and `bytei`. The current logic is actually clearer and easier to reason about so I am leaving it.